docs/standard/garbage-collection/implementing-dispose.md

```diff
@@ -77,7 +77,7 @@ The body of the method consists of three blocks of code:
 
   - **Managed objects that consume large amounts of memory or consume scarce resources.** Assign large managed object references to `null` to make them more likely to be unreachable. This releases them faster than if they were reclaimed non-deterministically.
 
-If the method call comes from a finalizer, only the code that frees unmanaged resources should execute. The implementer is responsible for ensuring that the false path doesn't interact with managed objects that may have been reclaimed. This is important because the order in which the garbage collector destroys managed objects during finalization is non-deterministic.
+If the method call comes from a finalizer, only the code that frees unmanaged resources should execute. The implementer is responsible for ensuring that the false path doesn't interact with managed objects that may have been disposed. This is important because the order in which the garbage collector disposes managed objects during finalization is non-deterministic.
```

Comment:
I'd also consider adding a bit more detail. Namely:

- All instances and their references are live during finalization (subsequent reclamation of the instances are non-deterministic as well, but IMO this is out of scope of this article and not really important since those references are just dead references waiting for GC to run)
- Instance (graph) may be revived (exactly once) during finalization - in most cases, this is bad practice to rely on this
- Exception thrown in a finalizer terminate the runtime (in default configuration), this includes `ObjectDisposedException` which some `Dispose` calls may throw (despite the fact they should be idempotent). This is the real danger of disposing already disposed instances.

Reply:
Ping @Maoni0 for review on this change. In particular, I wonder if using "dispose" in this paragraph could add confusion, as this is more about finalizers than Dispose methods.

Reply:
I was unsure about how much detail you wanted to put in. I basically wanted to start a discussion on this topic and find the right balance. But I personally expected a bit more detail when I read this article the first time. A link to another article or even a more technical blog post would also help.

But maybe this is just me needing to understand the topic in detail and not really that helpful for someone looking at it in a broader view.